app/models/foreman_tasks/task.rb

```diff
@@ -225,6 +225,31 @@ def to_label
       parts.join(' ').strip
     end
 
+    def self.get_latest_tasks_by_resource_ids(label, resource_ids, resource_type)
+      ids = resource_ids.join(',')
+      base_combined_table = search_for("label = #{label} and resource_type = #{resource_type} and resource_id ^ (#{ids})")
+
+      started_at = self::DynflowTask.arel_table[:started_at]
+      resource_id = ForemanTasks::Lock.arel_table[:resource_id]
+
+      grouped = base_combined_table.select(
+        "MAX(#{started_at}) AS started_at",
+        "#{resource_id} AS resource_id"
+      ).group(resource_id)
+
+      max_per_resource_id = self::DynflowTask.joins(
+        :locks
+      ).select(
+        "#{self::DynflowTask.table_name}.*, inner_select.resource_id"
+      ).joins(
+        "INNER JOIN (#{grouped.to_sql}) inner_select
+        ON inner_select.started_at = #{started_at}
+        AND inner_select.resource_id = locks_foreman_tasks_tasks.resource_id"
```

Comment:
I reckon this might be tricky, because of the very customized looking alias `locks_foreman_tasks_tasks`.
Is there a way to use arel for that?